tests/process.h

```diff
@@ -0,0 +1,118 @@
+#include "unistd.h"
+#include "wait.h"
+#include <cassert>
+#include <experimental/filesystem>
+#include <optional>
+#include <signal.h>
+
+namespace fs = std::experimental::filesystem;
+
+namespace h5s3::testing {
+
+class process {
+
+public:
+    using environment = std::vector<std::pair<std::string, std::string>>;
+
+    enum class state {
+        RUNNING,
+        EXITED,
+        SIGNALED,
+    };
+
+    class status {
+    public:
+        int code;
+        state process_state;
+    };
+
+    class error : public std::runtime_error {
+    public:
+        explicit error(const std::string& message) : std::runtime_error(message) {}
+    };
+
+private:
```

Comment:
Can we keep the visibility groups contiguous where possible?

Reply:
These were originally splt b/c I was using the `environment` typedef in the private section, but I wanted to make it public.  I'm not using that anymore, so I consolidate these.